cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go

```diff
@@ -12,3 +12,292 @@ limitations under the License.
+			if tc.statusCode == codes.OK {
+				opt1 := cmpopts.IgnoreUnexported(corev1.AvailablePackageSummary{}, corev1.AvailablePackageReference{}, corev1.Context{})
+				if got, want := availablePackageSummary, tc.expected; !cmp.Equal(got, want, opt1) {
+					t.Errorf("mismatch (-want +got):\n%s", cmp.Diff(want, got, opt1))
+				}
+			}
+		})
+	}
+}
+
+func TestGetAvailablePackageSummaries(t *testing.T) {
+	getter := func(context.Context) (dynamic.Interface, error) {
+		return fake.NewSimpleDynamicClientWithCustomListKinds(
+			runtime.NewScheme(),
+			map[schema.GroupVersionResource]string{
+				{Group: "foo", Version: "bar", Resource: "baz"}: "PackageList",
+			},
+		), nil
+	}
+	chartOK := &models.Chart{
+		Name:        "foo",
+		ID:          "foo",
+		Category:    "cat1",
+		Description: "best chart",
+		Icon:        "foo.bar/icon.svg",
+		Repo: &models.Repo{
+			Name:      "bar",
+			Namespace: "my-ns",
+		},
+		ChartVersions: []models.ChartVersion{
+			{Version: "1.0.0", AppVersion: "0.1.0"},
+			{Version: "1.0.0", AppVersion: "whatever"},
+			{Version: "whatever", AppVersion: "1.0.0"},
+		},
+	}
+	availablePackageSummaryOK := &corev1.AvailablePackageSummary{
+		DisplayName:      "foo",
+		LatestVersion:    "1.0.0",
+		IconUrl:          "foo.bar/icon.svg",
+		ShortDescription: "best chart",
+		AvailablePackageRef: &corev1.AvailablePackageReference{
+			Context:    &corev1.Context{Namespace: "my-ns"},
+			Identifier: "foo",
+		},
+	}
+	testCases := []struct {
+		name             string
+		charts           []*models.Chart
+		expectedPackages []*corev1.AvailablePackageSummary
+		statusCode       codes.Code
+	}{
+		{
+			name:             "it returns a set of availablePackageSummary from the database",
+			charts:           []*models.Chart{chartOK},
+			expectedPackages: []*corev1.AvailablePackageSummary{availablePackageSummaryOK},
+			statusCode:       codes.OK,
+		},
+		{
+			name:             "it returns an internal error status if response does not contain version",
+			charts:           []*models.Chart{{Name: "foo"}},
+			expectedPackages: []*corev1.AvailablePackageSummary{},
+			statusCode:       codes.Internal,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			mock, cleanup, manager := setMockManager(t)
+			defer cleanup()
+			s := Server{
+				clientGetter: getter,
+				manager:      manager,
+			}
+
+			rows := sqlmock.NewRows([]string{"info"})
+			rowCount := sqlmock.NewRows([]string{"count"}).AddRow(len(tc.charts))
+
+			for _, chart := range tc.charts {
+				chartJSON, err := json.Marshal(chart)
+				if err != nil {
+					t.Fatalf("%+v", err)
+				}
+				rows.AddRow(string(chartJSON))
+			}
+			mock.ExpectQuery("SELECT info FROM").
+				WillReturnRows(rows)
+
+			mock.ExpectQuery("^SELECT count(.+) FROM").
+				WillReturnRows(rowCount)
+
+			availablePackageSummaries, err := s.GetAvailablePackageSummaries(context.Background(), &corev1.GetAvailablePackageSummariesRequest{Context: &corev1.Context{}})
```

Comment:
Currently this is just testing only one case where the namespace is empty? We'd want to ensure that we test the cases:

- specific namespace results a 403 if the user doesn't have access to the ns
- specific namespace results in a query for charts in that ns only otherwise,
- blank namespace results in a query for charts in the kubeapps namespace and any namespace to which the user has access

I think?